### PR TITLE
Implement mentee application windows shopping feature

### LINKED ISSUE #94: Mentor window shopping
**Is your feature request related to a problem? Please describe.**
* Users should be able to view the mentor list of a program without the necessity to login.

**Describe the solution you'd like**
* Make `get mentors` API public.
* If a non logged-in user tries to apply to a mentor they should be redirected to login page.
* It's nice to have a log-in button on top right hand corner.

### LINKED ISSUE #103: Restrict unauthorised users from accessing the authorised UIs
**Is your feature request related to a problem? Please describe.**
- Currently, Users can access the authorised UI's without being authorised. If the user trying to access a UI that needs to be authorised the user should be redirected to the sign-in page

**Describe the solution you'd like**
- Check whether the user has logged in or not when the user trying to access the UIs that requires to be authorised.

### LINKED ISSUE #109: Replace the user avatar in the admin dashboard with the user profile image
**Is your feature request related to a problem? Please describe.**
The user avatar in the admin dashboard navbar hasn't still replaced with the user profile image. 
image

**Describe the solution you'd like**
* Import the `UserContext` context 
* Remove the current icon from the avatar
* Put a new `src` attribute to the `Avatar` element with the value of the image URL of the current user

### PR DESCRIPTION
**Pull request by Gravewalker666**
## Purpose

The purpose of this PR is to fix #94, fix #109, fix #103 
- Prompt users with a modal to sign in where sign-in is necessary.
- Prevent users from accessing unauthorised pages.
- Replace the avatar icon with the profile picture in the dashboard navigation bar.

## Goals

To implement the windows shopping feature and prevent users from accessing unauthorised pages.

#### Window Shopping feature 
- Created a new `LogInModal` component high up in the component hierarchy(`/src`)
- Provided public users with a button to click on and get into the mentor list 
- Configured the above-mentioned modal to pop up when the user clicks on a mentor

https://user-images.githubusercontent.com/45477334/115137769-acf72e00-a045-11eb-846e-d91b132845b8.mp4

#### Preventing users from accessing unauthorised pages
- Used the above-defined `LogInModal` and prompted the non-logged in users with the option to sign in to continue
- Made the `LogInModal` un-cancellable 
image
image

#### Replace the avatar icon with the profile picture
- Removed the avatar icon and replace it with `src={user?.imageUrl}`
image

##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [x] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

## Test environment
- webpack dev server
- Safari
- macOS Big Sur

### CODE REVIEW

File: src/components /LogInModal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { LogInModalProps } from './interfaces';
+import { Modal } from 'antd';
+
+function LogInModal(props: LogInModalProps) {
+  const handleOk = () => {
+    window.location.href =
+      'http://localhost:8080/oauth2/authorization/linkedin';
```
Comment (anjula-sack) [inline src/components /LogInModal/index.tsx]: Let's create another variable to store the sign in url

File: src/scenes/Home/scenes/RequestMentors/components/Mentors/index.tsx [hunk 2]
```diff
@@ -68,21 +84,39 @@ function Mentors() {
...
+                        <Row justify="center">
+                          <Avatar size={64} src={item.profile.imageUrl} />
+                          <Title level={4} className={styles.cardTitle}>
+                            {item.profile.firstName} {item.profile.lastName}
+                          </Title>
+                          <Paragraph>{item.profile.headline}</Paragraph>
+                        </Row>
+                      </Col>
+                    </Row>
+                  </Card>
```
Comment (anjula-sack) [inline src/scenes/Home/scenes/RequestMentors/components/Mentors/index.tsx]: This is the mentor card right? It's better to create another component for this.

**Review by anjula-sack — CHANGES_REQUESTED** [review]
Awesome! @Gravewalker666

File: src/components /LogInModal/index.tsx (hunk 1 above)
Comment (anjula-sack) [reply src/components /LogInModal/index.tsx]: BTW, We are going to pass the redirect location URL right?

Comment (Gravewalker666) [reply src/components /LogInModal/index.tsx]: > Let's create another variable to store the sign-in URL

Sure, On it

> BTW, We are going to pass the redirect location URL right?

Let's do that part after getting this PR merged, we have to handle the redirection in some other places as well so let's work on the whole bunch together in a different PR

File: src/scenes/Home/scenes/RequestMentors/components/Mentors/index.tsx (hunk 2 above)
Comment (Gravewalker666) [reply src/scenes/Home/scenes/RequestMentors/components/Mentors/index.tsx]: Yeah, that's better, thanks for the suggestion.

Comment (Gravewalker666) [reply src/scenes/Home/scenes/RequestMentors/components/Mentors/index.tsx]: Resolved with the commit [b04eb4d](https://github.com/sef-global/scholarx-frontend/pull/111/commits/b04eb4d)

Please review =)

File: src/components /LogInModal/index.tsx (hunk 1 above)
Comment (Gravewalker666) [reply src/components /LogInModal/index.tsx]: Resolved with the commit [b04eb4d](https://github.com/sef-global/scholarx-frontend/pull/111/commits/b04eb4d)

Please review =)

**Comment by anjula-sack** [comment]
The build is failing and why is there a change in the `package-lock.json`?

**Review by anjula-sack — APPROVED** [review]
Good job! @Gravewalker666

File: src/components /LogInModal/index.tsx (hunk 1 above)
Comment (Gravewalker666) [reply src/components /LogInModal/index.tsx]: b04eb4d

**Pull request merged**